tobac/segmentation.py

```diff
@@ -588,7 +589,7 @@ def segmentation(
 
     for i, field_i in enumerate(field_time):
         time_i = field_i.coord("time").units.num2date(field_i.coord("time").points[0])
-        features_i = features.loc[features["time"] == time_i]
+        features_i = features.loc[features["time"] == np.datetime64(time_i)]
```

Comment:
Yes, it fixes cases where the Pandas -> Xarray -> Iris conversion of times produces two functionally equivalent times from `cf-units` but do not evaluate as equal. This will probably not be necessary with the conversion to xarray everywhere. I believe that it shouldn't change behavior otherwise.

Reply:
We should make a list of all the iris -> xarray oddities to make sure we can cover them with conversion decorators for the v1.6 release